app/views/posts/show.html.erb

```diff
@@ -1,10 +1,20 @@
-<button class="back-link"><%= link_to "Go to all posts", :back %></button>
+<button class="back-link"><%= link_to "Go to all posts", user_posts_path(@user.id) %></button>
 
 <div class="section">
   <div class="section-post">
+    <div class="title-flex">
       <h1 class="section-post__title"><%= @post.title %> by <%= @user.name %></h1>
-      <p class="section-post__counter">Comments: <%= @post.comments_counter %> | Likes: <%= @post.likes_counter %></p>
-      <p class="section-post__text"><%= @post.text %></p>
+      <div class="title-flex">
+        <button class="back-link"><%= link_to "Comment", new_user_post_comment_path(@user.id, @post.id) %></button>
+        
+        <%= form_with url: user_post_likes_url(@user, @post), method: 'post' do |f| %>
+          <%= f.submit 'Like', class: "back-link" %>
+        <% end %>
+      </div>
+    </div>
```

Comment:
- [ ] An error occurs when trying to go to the `/posts` index page and `/posts/:id` show page as shown below 👇 Please fix this bug 🐛 
![image](https://user-images.githubusercontent.com/46839548/192046811-a7c4ab9a-f3db-4447-9a72-2b4c870df28e.png)

Reply:
Hi, @codepantha Thanks for your review 😁 

Everything seems to be working on my side with the current code. Any idea of why? 🤔 

![image](https://user-images.githubusercontent.com/97407840/192049368-0460b483-edff-49c7-8b03-1ad1086d19bf.png)

Reply:
I think in the course of developing the project, you probably have some data you have generated either through rails console or any other method that masked the issue, and as a result, you haven't noticed this problem.

Reply:
@codepantha Thanks! I refactored the post controller `show` and `index` methods. The project works while having information in the database.